### PR TITLE
Support ipvlan and ipvtap configurations in nixos/networkd

### PR DESCRIPTION
**Pull request by philiptaron**
## Description of changes

`systemd-networkd` supports creating and configuring IPVLAN and IPVTAP devices. This PR introduces NixOS module support for those devices in the form of adding the configuration options to `systemd.network.netdevs`.

This is a different direction than https://github.com/NixOS/nixpkgs/pull/48753 went.

## Things done

I tested this out with my personal NixOS configuration. 

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/system/boot/networkd.nix [hunk 1]
```diff
@@ -191,6 +202,10 @@ let
         (assertValueOneOf "ReorderHeader" boolValues)
       ];
 
+      sectionIPVLAN = checkUnitConfig "IPVLAN" ipVlanVtapChecks;
+
+      sectionIPVTAP = checkUnitConfig "IPVLAN" ipVlanVtapChecks;
```
Comment (RaitoBezarius) [inline nixos/modules/system/boot/networkd.nix:207]:
```suggestion
      sectionIPVTAP = checkUnitConfig "IPVTAP" ipVlanVtapChecks;
```

**Review by RaitoBezarius — CHANGES_REQUESTED** [review]
Small mistake, but looks good to me.

File: nixos/modules/system/boot/networkd.nix (hunk 1 above)
Comment (philiptaron) [reply nixos/modules/system/boot/networkd.nix]: 🤦🏻 Thanks. Fixed.

**Review by RaitoBezarius — APPROVED** [review]

**Comment by Artturin** [comment]
Does this replace https://github.com/NixOS/nixpkgs/pull/48753, can tests be added like it has.

**Comment by Mic92** [comment]
Tests in the other one are old enough to be still in perl.

**Pull request merged**